src/App.tsx

```diff
@@ -17,18 +59,27 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter handleFiltrationQueries={handleFiltrationQueries} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0 && <Loader />}
```

Comment:
You should show the loader only if you are loading something. If the API returns an empty array the user sees an infinite loader